File: Energi-Potensial/Energi-Potensial.py
gravitasi = 10

def main ():
    pilihan = 0
    massa = 0
    ketinggian = 0
    energi_potensial = 0

    print ("========== Penghitung Energi Potensial ==========")

    while True:
        print ("1. Mencari energi potensial")
        print ("2. Mencari massa")
        print ("3. Mencari ketinggian")
        print ("4. Keluar")
        pilihan = int (input ("Pilih salah satu: "))

        if pilihan == 1:
            energi_potensial = hasil_energi_potensial ()
            print (f"Energi potensial: {energi_potensial} Joule")
            break
        elif pilihan == 2:
            massa = hasil_massa ()
            print (f"Massa: {massa} kg")
            break
        elif pilihan == 3:
            ketinggian = hasil_ketinggian ()
            print (f"Ketinggian: {ketinggian} m")
            break
        elif pilihan == 4:
            print ("Terima kasih telah menggunakan program ini")
            break
        else:
            print ("Input error")

def hasil_energi_potensial ():
    massa = float (input( "Masukkan massa (satuan kg): "))
    ketinggian = float (input ("Masukkan ketinggian (satuan m): "))
    return massa * gravitasi * ketinggian

def hasil_massa ():
    energi_potensial = float (input ("Masukkan energi potensial (satuan Joule): "))
    ketinggian = float (input ("Masukkan ketinggian (satuan m): "))
    return energi_potensial / (gravitasi * ketinggian)

def hasil_ketinggian ():
    energi_potensial = float (input ("Masukkan energi potensial (satuan Joule): "))
    massa = float (input ("Masukkan massa (satuan kg): "))
    return energi_potensial / (massa * gravitasi)

if __name__ == "__main__":
    main ()

File: Energi-Potensial/Energi-Potensial.cpp
#include <iostream>

const float gravitasi = 10;
float hasil_massa ();
float hasil_ketinggian ();
float hasil_energi_potensial ();

int main ()
{
    unsigned short pilihan = 0;
    float massa = 0, ketinggian = 0, energi_potensial = 0;

    std::cout << "========== Penghitung Energi Potensial ==========\n";

    do
    {
        std::cout << "1. Mencari energi potensial\n";
        std::cout << "2. Mencari massa\n";
        std::cout << "3. Mencari ketinggian\n";
        std::cout << "4. Keluar\n";
        std::cout << "Pilih salah satu: ";
        std::cin >> pilihan;

        switch (pilihan)
        {
            case 1:
                energi_potensial = hasil_energi_potensial ();
                std::cout << "Energi potensial: " << energi_potensial << " Joule";
                break;
            case 2:
                massa = hasil_massa ();
                std::cout << "Massa: " << massa << " kg";
                break;
            case 3:
                ketinggian = hasil_ketinggian ();
                std::cout << "Ketinggian: " << ketinggian << " m";
                break;
            case 4:
                std::cout << "Terima kasih telah menggunakan program ini";
                break;
            default:
                std::cout << "Input error\n";
                break;
        }
    } while (pilihan < 1 || pilihan > 4);
    
    return 0;
}

float hasil_energi_potensial ()
{
    float massa = 0, ketinggian = 0;

    std::cout << "Masukkan massa (satuan kg): ";
    std::cin >> massa;

    std::cout << "Masukkan ketinggian (satuan m): ";
    std::cin >> ketinggian;

    return massa * gravitasi * ketinggian;
}

float hasil_massa ()
{
    float energi_potensial = 0, ketinggian = 0;

    std::cout << "Masukkan energi potensial (satuan Joule): ";
    std::cin >> energi_potensial;

    std::cout << "Masukkan ketinggian (satuan m): ";
    std::cin >> ketinggian;

    return energi_potensial / (gravitasi * ketinggian);
}

float hasil_ketinggian ()
{
    float energi_potensial = 0, massa = 0;

    std::cout << "Masukkan energi potensial (satuan Joule): ";
    std::cin >> energi_potensial;

    std::cout << "Masukkan massa (satuan kg): ";
    std::cin >> massa;

    return energi_potensial / (massa * gravitasi);
}